app/views/shared/_order_detail_cell.html.haml

```diff
@@ -10,11 +10,12 @@
     - if od.product.offline?
       = tooltip_icon "icon-warning-sign", t("instruments.offline.note")
 
-  - if od.survey_completed?
-    .order-detail-extra= link_to 'View Order Form', od.external_service_receiver.show_url, :target => '_blank'
-  - od.stored_files.template_result.each do |stored_file|
-    .order-detail-extra
-      = link_to t("order_details.show.link.view.order_file"), facility_template_result_path(stored_file)
+  - if can?(:template_results, od)
```

Comment:
It's a little tricky to find a permission that works for any external_service_receiver, but I figure if they shouldn't be able to see `template_results`, they probably shouldn't be able to see order forms either. Let me know if there is a better solution.